server/src/main/java/com/lunchtime/controllers/RestaurantResource.java

```diff
@@ -0,0 +1,97 @@
+package com.lunchtime.controllers;
+
+import com.lunchtime.models.Restaurant;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.lunchtime.service.RestaurantService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+
+@RestController
+@RequestMapping("/api/restaurants")
+public class RestaurantResource {
+
+    @Autowired
+    private RestaurantService restaurantService;
+
+    @PostMapping
+    public ResponseEntity<Restaurant> create(@Valid @RequestBody Restaurant restaurant) throws URISyntaxException {
+        if (restaurant.getId() != null) {
+            return ResponseEntity.badRequest()
+                .build();
+        }
+
+        Restaurant result = restaurantService.save(restaurant);
+        return ResponseEntity.created(new URI("/api/restaurants"))
+            .body(result);
+    }
+
```

Comment:
I think you meant `if (restaurant.getId() == null)` in line 33

Reply:
No if new Restaurant has id (restaurant.getId() != null) then user must use update request, because client can't know id in new Restaurant entity. Only database creates id in every new entities

Reply:
Got it. Thanks!